### PR TITLE
Momentum Based Gradient Optimizer and Optimization Techniques for Gradient Descent

### LINKED ISSUE #4527: [Machine Learning] Momentum-based Gradient Optimizer and Optimization techniques for Gradient Descent
#### Proposed title of article
Momentum-based Gradient Optimizer and Optimization techniques for Gradient Descent

#### Proposed article introduction
Gradient Descent is an iterative optimization algorithm that finds the smallest value for a function. The general idea is to set the parameters to random values and then iterate in small steps in the direction of the `slope`. Gradient descent is widely used in supervised learning to minimize the error function and find the best parameter values.
Adaptive Optimization (A.O.) Algorithms are becoming more popular due to their ability to converge quickly. In contrast to the traditional Gradient Descent algorithm, all of these algorithms use statistics from previous iterations to strengthen the convergence process. We will be discussing it in detail in this article.
#### Key takeaways 
- Momentum-based Optimization
- Extensions  designed for gradient descent algorithm
- How does momentum help in gradient descent
- Optimization techniques
#### References
N/A

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by NdoloBonface**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

`This closes #4527 

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by NdoloBonface** [comment]
Hello @lalith1403 
Could you please look at this PR, please.

**Comment by Kayere** [comment]
Plagiarism check done ✔️

File: content/articles/momentum-based-gradient-optimizer-and-optimization-techniques-for-gradient-descent/index.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+- [Optimizer to use](#optimizer-to-use)
+
+### Optimization techniques
+For a brief while, we'll consider various optimization methods frequently employed in Deep Learning. In other words, if you came to this article hoping to learn the equation and mathematical explanation, don't bother. 
+
+This page also includes a quick overview of the many approaches that are accessible, including those provided by the Keras package.
+
+### Momentum-based Optimization
+
+Utilize an Adaptive Optimization Algorithm (AOA) that uses exponentially-weighted averaging of gradients from prior rounds to keep convergence from deviating. Most real-world applications, such as deep neural networks, rely on noisy input to train.
```
Comment (lalith1403) [inline content/articles/momentum-based-gradient-optimizer-and-optimization-techniques-for-gradient-descent/index.md:46]: Can you explain AOA before you introduce it?

File: content/articles/momentum-based-gradient-optimizer-and-optimization-techniques-for-gradient-descent/index.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+This page also includes a quick overview of the many approaches that are accessible, including those provided by the Keras package.
+
+### Momentum-based Optimization
+
+Utilize an Adaptive Optimization Algorithm (AOA) that uses exponentially-weighted averaging of gradients from prior rounds to keep convergence from deviating. Most real-world applications, such as deep neural networks, rely on noisy input to train. 
+
+There must be a systematic approach to feeding the data to minimize the impact of noise during optimization. To resolve this issue, exponentially weighted averages might be used.
+
+Using an iterative process, we can approximate the trends in a P-dimensional noisy dataset by calculating the parameters as we go through the data:
+```
```
Comment (lalith1403) [inline content/articles/momentum-based-gradient-optimizer-and-optimization-techniques-for-gradient-descent/index.md:51]: Use latex to write mathematical equations.

File: content/articles/momentum-based-gradient-optimizer-and-optimization-techniques-for-gradient-descent/index.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+On iteration p:
+    Get next  θp 
+    v_θ = β vθ + (1 - β)θp
+```
+**Vθ** is averaged across its preceding **1/(1 - β)** iterations, giving a value of **vθ**. With the help of averages, you can eliminate all the randomness and merely keep track of the overall trend. As a result of this method, the algorithm is more resistant to sample noise and can be trained more quickly.
+### How does momentum help in gradient descent
+To describe a gradient descent algorithm, the term `momentum` is used. This method works by detecting the direction of the steepest slope in its present state and then travelling in the direction of that sharpest slope to update its status.
```
Comment (lalith1403) [inline content/articles/momentum-based-gradient-optimizer-and-optimization-techniques-for-gradient-descent/index.md:58]: This is factually incorrect. Momentum is not used to describe the algorithm, it is additional information used to converge faster.

File: content/articles/momentum-based-gradient-optimizer-and-optimization-techniques-for-gradient-descent/index.md [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+Using Stochastic Graduated Descent instead of Classic Gradient Descent, several of the flaws in the classic method are corrected By computing the derivative for each point on the trajectory it follows. It attempts to make up for its lack of computational rigour. It has more considerable noise than Gradient Descent. Hence it takes longer iterations to get to a minimal solution using SGD. In contrast to Gradient Descent, SGD computes the derivatives for all points at the same time.
+### Adaptive Optimization Algorithm techniques.
+In the next section, we'll take a look at a few approaches used in AOAs.
+### Adaptive Moment Estimation
+RMSProp and momentum combine to create this effect. Use this method to figure out how fast your brain is adapting to new information. Like momentum, it saves the declining arithmetic mean of the squared gradients and an average of the past gradients, which is a similar feature.
+
+On a smooth surface, it's considerably easier for Adam to make a mistake because of his high coefficient of friction. So, for example, this strategy's pseudocode appears like this:
+```
+D = 0
```
Comment (lalith1403) [inline content/articles/momentum-based-gradient-optimizer-and-optimization-techniques-for-gradient-descent/index.md:74]: use latex

File: content/articles/momentum-based-gradient-optimizer-and-optimization-techniques-for-gradient-descent/index.md [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+    D = β1K + (1 -β1) dW
+     D = V⁄{1 -β1i}
+    K = β2K + (1 -β2) dW2
+    K = S⁄{1 -β2i}
+    W = W - α V⁄√S + ε
+```
+
+### RMSProp
+
+Improve AdaGrad by applying the adaptive learning rate method based on the Root Mean Square. The exponential moving average is used rather than the cumulative sum of squared gradients used by AdaGrad There is only one step that differs between AdaGrad and RMSProp: the first one.
```
Comment (lalith1403) [inline content/articles/momentum-based-gradient-optimizer-and-optimization-techniques-for-gradient-descent/index.md:87]: Needs more explanation

**Review by lalith1403 — CHANGES_REQUESTED** [review]
Many changes are required. This is a good article but needs more information. Please add more info where necessary. I have highlighted a few areas that need additional info. However, there are many more places that would benefit from a detailed explanation.

**Comment by NdoloBonface** [comment]
@lalith1403 Done 
Thanks.

File: content/articles/momentum-based-gradient-optimizer-and-optimization-techniques-for-gradient-descent/index.md (hunk 4 above)
Comment (lalith1403) [reply content/articles/momentum-based-gradient-optimizer-and-optimization-techniques-for-gradient-descent/index.md:70]: This is not latex. Latex equations begin with $ and end with $. Please check published articles with equations in them and update accordingly.

**Review by lalith1403 — CHANGES_REQUESTED** [review]
Check published articles for latex references. Thank you.

**Comment by lalith1403** [comment]
I will be closing this PR now. Please research the latex part and submit a new PR.

**Comment by NdoloBonface** [comment]
@lalith1403 I did some changes. Some of the code did not render when I tried to view it. Thanks.

**Comment by hectorkambow** [comment]
> I will be closing this PR now. Please research the latex part and submit a new PR.

This was the correct move - if the PR still needs more revisions. 

Close it and a new PR will be needed.

👍  

@lalith1403
